lib/MumbleClient.js

```diff
@@ -533,6 +533,16 @@ MumbleClient.prototype._userRemove = function( data ) {
         user._detach();
         delete this.users[ user.id ];
         delete this.sessions[ data.session ];
+
+        // Also remove the user from the _users array
+        var newUsers = [];
+        for( var i = 0; i < this._users.length; i += 1 ) {
+            var oldUser = this._users[ i ];
+            if( oldUser.id !== user.id ) {
+                newUsers.push( oldUser );
+            }
+        }
+        this._users = newUsers;
```

Comment:
Instead of replacing the whole array, I get the feeling it would be better to just delete the user from the existing one.

```javascript
for( var i = 0; i < this._users.length; /* no increment on purpose */ ) {

    // Check the user in the current index.
    //
    // We'll either remove an item from the array or increment i. Both of
    // these actions result in 'i' moving one position closer to the end which
    // guarantees the loop termination.
    var current = this._users[ i ];
    if( current.id === user.id ) {
         this._users.splice( i, 1 ); 
    } else {
        i += 1;
    }
}
```

Mainly I'm basing this on the fact that at some point we might end up keeping a reference to the `_users` list somewhere else and at that point I feel it might be safer to have only one such list around.

Although I do see this might also be a bad idea in some other ways so do let me know how you feel about this!

Reply:
I looped backwards in a for loop to do this, but otherwise our approach is very similar. Take a look and tell me what you think about my approach.

Reply:
That works great! Thanks for the change!

Also sorry for missing the last two weeks on Github. :)